math/aliquot_sum.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file
+ * \brief Program to return the [Aliquot Sum](https://en.wikipedia.org/wiki/Aliquot_sum) of a number
+ * 
+ * \details
+ * The Aliquot sum s(n) of a non-negative integer n is the sum of all
+ * proper divisors of n, that is, all the divisors of n, other than itself.
+ * For example, the Aliquot sum of 18 = 1 + 2 + 3 + 6 + 9 = 21
+ */
+
+#include <stdexcept>
+#include <cassert>
+
+/**
+ * Function to return the aliquot sum of a number
+ * @param num The input number
+ */
+int aliquot_sum(int num) {
+	if(num < 0) {
+        throw std::invalid_argument("Received negative value");
```

Comment:
instead of throwing an expection on negative numbers you can enforce positive values by the type of the interface, why not taking unsigned long long int as argument?

Reply:
Now that I think about it, this seems way more logical. I didn't realise it earlier :sweat_smile: